### DISCUSSION TITLE
A 'Density' is not a registered object

### DISCUSSION THREAD

**Discussion by tctheodosiou** [idaholab/moose#25191] [Q&A Modules: General] [answered]
Hello,
I am trying to run the Thermal Mortar Contact problem (https://mooseframework.inl.gov/modules/combined/tutorials/introduction/step01.html)

I get the following error message:
_*** ERROR ***
A 'Density' is not a registered object._

I included TensorMechanics and HeatConduction modules. After searching the documentation, I noticed that Density in mentioned a lot in the "Combined module", so I included it as well. Recompiled with make clobberall, and then make. Still Density is not working. 
Should I assume that I should include some other module too? Which? (And how do I determine which modules are needed for each App? e.g. Re. Density, it is not that obvious where to find it).

(I even reinstall everything dev & prebuilt to make sure that everything is compiled correctly!!!)

By the way, the make file didn't have any reference to the Combined module, so I added "Combined := yes" by hand. Shouldnt it be in the list when the Makefile was created by stork?

Thank you for your time!

**Comment by GiudGiud** [comment, marked as answer]
Hello
Density is part of the miscellaneous module

To find where an object comes from, you can do this in moose/:
```
find . | grep "Density.C"
```
or you can use various search features. Or just use the combined module. 

There is a "ALL_MODULES" boolean in the Makefile you could have used

Guillaume

**Reply by tctheodosiou** [reply]
Hi Guillaume,
Thank you again for your quick support! Your solution worked (as expected).

Re. ALL_MODULES, I am aware of that, but in the documentation, it says to avoid all modules to include only what's necessary. (In a worst case scenario, I would enable all modules. But for the moment, I 's like to "follow the rules" :) ) In the next case of "missing object", I will search for it with find, as you suggested. (Enabling the Combined module, did not solve the problem).

Sorry for my ignorant questions. MOOSE is a huge framework, and I am trying to adapt.

**Reply by GiudGiud** [reply]
ah yes I would not recommend all_modules, slows down building quite a bit
no worries
